<launch>
    <arg name="rviz" default="true"/>
    <arg name="steps_per_phase" default="10"/>
    <arg name="plotting" default="false" />
    <arg name="file_write_cb" default="false" />
    <arg name="box_side" default="0.2"/>
    <arg name="box_x" default="0.15"/>
    <arg name="box_y" default="0.4"/>
    <arg name="box_parent_link" default="workcell_base"/>
    <arg name="testing" default="false"/>

    <!-- Load Robot Model    -->
    <param name="robot_description"  command="$(find xacro)/xacro --inorder $(find trajopt_examples)/urdf/pick_and_place_plan.xacro"/>

    <!-- Load Robot Semantic -->
    <param name="robot_description_semantic" textfile="$(find trajopt_examples)/config/pick_and_place_plan.srdf" />

    <!-- Set simulated box parameters -->
    <param name="box_side" value="$(arg box_side)"/>
    <param name="box_x" value="$(arg box_x)"/>
    <param name="box_y" value="$(arg box_y)"/>
    <param name="box_parent_link" value="$(arg box_parent_link)"/>

    <!-- Launch planner node -->
    <group unless="$(arg testing)" >
      <node name="pick_and_place_plan_node" pkg="trajopt_examples" type="trajopt_examples_pick_and_place_plan" output="screen">
          <param name="steps_per_phase" value="$(arg steps_per_phase)"/>
          <param name="plotting" value="$(arg plotting)"/>
          <param name="file_write_cb" value="$(arg file_write_cb)" />
          <param name="testing" value="$(arg testing)" />
      </node>

      <!-- Launch visualization -->
      <node if="$(arg rviz)" pkg="rviz" type="rviz" name="pick_and_place_viz"
          args="-d $(find trajopt_examples)/config/pick_and_place_plan.rviz" />
    </group>

    <!-- For unit testing. Not needed for regular operation -->
    <test if="$(arg testing)" pkg="trajopt_examples" type="trajopt_examples_pick_and_place_plan" test-name="trajopt_examples_pick_and_place_plan_test" time-limit="120.0">
      <param name="steps_per_phase" value="$(arg steps_per_phase)"/>
      <param name="plotting" value="false"/>
      <param name="file_write_cb" value="false" />
      <param name="testing" value="true" />
    </test>
</launch>
